src/service/alert.service.ts

```diff
@@ -2,11 +2,37 @@ import { Injectable, Sanitizer, SecurityContext } from '@angular/core';
 import { TranslateService } from 'ng2-translate';
 import { ConfigHelper } from '../helper';
 
+export type AlertType =  'success' | 'danger' | 'warning' | 'info';
+
+export class Alert {
+    id: number;
+    type: AlertType;
+    msg: string;
+    params?: any;
+    timeout: number;
+    toast: boolean;
+    position: string;
+    scoped: boolean;
+    close: (alerts: Alert[]) => void;
+}
+
+export class AlertOptions {
```

Comment:
This class can extend Alert and add just alertID I guess or all common items can be in BaseAlert and both Alert and AlertOptions can extend that. I also need to check if alertId is really required may be we can just use one Alert class